synapse/types/__init__.py

```diff
@@ -979,3 +980,40 @@ class UserProfile(TypedDict):
 class RetentionPolicy:
     min_lifetime: Optional[int] = None
     max_lifetime: Optional[int] = None
+
+
+class TaskStatus(str, Enum):
+    """Status of a scheduled task"""
+
+    # Task is scheduled but not active
+    SCHEDULED = "scheduled"
+    # Task is active and probably running, and if not
+    # will be run on next scheduler loop run
+    ACTIVE = "active"
+    # Task has completed successfully
+    COMPLETE = "complete"
+    # Task is over and either returned a failed status, or had an exception
+    FAILED = "failed"
+
+
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class ScheduledTask:
+    """Description of a scheduled task"""
+
+    # id used to identify the task
+    id: str
+    # name of the action to be run by this task
+    action: str
+    # current status of this task
+    status: TaskStatus
+    # if the status is SCHEDULED then this represents when it should be launched,
+    # otherwise it represents the last time this task got a change of state
+    timestamp: int
```

Comment:
This is...in milliseconds since the epoch? Does it have a timezone?

Reply:
It is compared to `time_msec` which is `the current system time in milliseconds since epoch`, so on a properly configured system it should be UTC for Linux and MacOS at least. I am not sure for Windows but I am not sure we want to care either :innocent: .

I'll complete the comment.